src/Client/Web/Bit.Client.Web.BlazorUI.Tests/Inputs/BitTextFieldTests.cs

```diff
@@ -0,0 +1,267 @@
+
+            Assert.AreEqual("Text", bitTextField.FirstElementChild.GetAttribute("type"));
+            Assert.IsTrue(bitTextFieldRevealPassword.FirstElementChild.ClassList.Contains($"bit-icon--Hide"));
+        }
+
+        [DataTestMethod,
+            DataRow(true, false),
+            DataRow(true, true),
+            DataRow(false, false),
+            DataRow(false, true),
+        ]
+        public void BitTextFieldMustRespondToTheClickEvent(bool isEnabled, bool isMultiLine)
+        {
+            var component = RenderComponent<BitTextFieldTest>(
+                parameters =>
+                {
+                    parameters.Add(p => p.IsEnabled, isEnabled);
+                    parameters.Add(p => p.IsMultiLine, isMultiLine);
+                });
+
+            var bitTextField = isMultiLine ? component.Find(".bit-txt textarea") : component.Find(".bit-txt input");
+
+            bitTextField.Click();
+
+            Assert.AreEqual(isEnabled ? 1 : 0, component.Instance.CurrentCount);
+        }
+
+        [DataTestMethod,
+            DataRow(true, false),
+            DataRow(true, true),
+            DataRow(false, false),
+            DataRow(false, true),
+        ]
+        public void BitTextFieldMustRespondToTheFocusEvent(bool isEnabled, bool isMultiLine)
+        {
+            var component = RenderComponent<BitTextFieldTest>(
+                parameters =>
+                {
+                    parameters.Add(p => p.IsEnabled, isEnabled);
+                    parameters.Add(p => p.IsMultiLine, isMultiLine);
+                });
+
+            var bitTextField = isMultiLine ? component.Find(".bit-txt textarea") : component.Find(".bit-txt input");
+
+            bitTextField.Focus();
+            Assert.AreEqual(isEnabled ? 1 : 0, component.Instance.CurrentCount);
+
+            bitTextField.FocusIn();
+            Assert.AreEqual(isEnabled ? 2 : 0, component.Instance.CurrentCount);
+
+            bitTextField.FocusOut();
+            Assert.AreEqual(isEnabled ? 3 : 0, component.Instance.CurrentCount);
+        }
+
+        [DataTestMethod,
+            DataRow(true, false),
+            DataRow(true, true),
+            DataRow(false, false),
+            DataRow(false, true),
+        ]
+        public void BitTextFieldMustRespondToTheKeyEvent(bool isEnabled, bool isMultiLine)
+        {
+            var component = RenderComponent<BitTextFieldTest>(
+                parameters =>
+                {
+                    parameters.Add(p => p.IsEnabled, isEnabled);
+                    parameters.Add(p => p.IsMultiLine, isMultiLine);
+                });
+
+            var bitTextField = isMultiLine ? component.Find(".bit-txt textarea") : component.Find(".bit-txt input");
+
+            bitTextField.KeyDown("down");
+            Assert.AreEqual(isEnabled ? 1 : 0, component.Instance.CurrentCount);
+
+            bitTextField.KeyUp("up");
+            Assert.AreEqual(isEnabled ? 2 : 0, component.Instance.CurrentCount);
```

Comment:
please use different properties to test different keyboard events

Reply:
done